src/Module/OStatus/PubSub.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\OStatus;
+
+use Friendica\App;
+use Friendica\Core\L10n;
+use Friendica\Core\Protocol;
+use Friendica\Core\System;
+use Friendica\Database\Database;
+use Friendica\Model\Contact;
+use Friendica\Model\GServer;
+use Friendica\Model\Post;
+use Friendica\Module\Response;
+use Friendica\Network\HTTPException;
+use Friendica\Protocol\OStatus;
+use Friendica\Util\Network;
+use Friendica\Util\Profiler;
+use Friendica\Util\Strings;
+use Psr\Log\LoggerInterface;
+
+class PubSub extends \Friendica\BaseModule
+{
+	/** @var Database */
+	private $database;
+
+	public function __construct(Database $database, L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, array $server, array $parameters = [])
+	{
+		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
+
+		$this->database = $database;
+	}
+
+	protected function post(array $request = [])
+	{
+		$xml = Network::postdata();
+
+		$this->logger->info('Feed arrived from ' . $this->server['REMOTE_ADDR'] . ' for ' . $this->args->getCommand() . ' with user-agent: ' . $this->server['HTTP_USER_AGENT']);
```

Comment:
Include src\App\Request in this class and than:

```suggestion
		$this->logger->info('Feed arrived.', ['from' =>  . $this->request->getRemoteAddress(), ' for ' => $this->args->getCommand(), 'user-agent' => $this->server['HTTP_USER_AGENT']]);
```